prow/jenkins/jenkins.go

```diff
@@ -311,6 +312,55 @@ func (c *Client) ListBuilds(jobs []string) (map[string]JenkinsBuild, error) {
 	return jenkinsBuilds, nil
 }
 
+// ListBuildsParallel returns a list of all Jenkins builds for the
+// provided jobs (both scheduled and enqueued). Listing of builds
+// happens in parallel.
+func (c *Client) ListBuildsParallel(jobs []string) (map[string]JenkinsBuild, error) {
+	// Get queued builds.
+	jenkinsBuilds, err := c.GetEnqueuedBuilds(jobs)
+	if err != nil {
+		return nil, err
+	}
+
+	buildChan := make(chan map[string]JenkinsBuild, len(jobs))
+	errChan := make(chan error, len(jobs))
+	wg := &sync.WaitGroup{}
+	wg.Add(len(jobs))
+
+	// Get all running builds for all provided jobs.
+	for _, job := range jobs {
+		// Start a goroutine per list
+		go func(job string, buildChan chan<- map[string]JenkinsBuild, errChan chan<- error) {
```

Comment:
nit: `buildChan` and `errChan` don't need to be parameters to the goroutine, they will be captured appropriately.